tools/check_package_release.py

```diff
@@ -0,0 +1,58 @@
+"""Determine if new package should be release (based on version) and consequently write information to env."""
+import os
+import sys
+
+import requests
+
+VERSION_MATCHSTR = r'\s*__version__\s*=\s*"(\d+\.\d+\.\d+)"'
+RESPONSE_ERROR = 404
+
+
+def _check_no_version_pypi(pypi_endpoint, package_name, package_version):
+    response = requests.get(pypi_endpoint, timeout=10)
+    if response.status_code == RESPONSE_ERROR:
+        # Version doesn't exist on Pypi - do release
+        print(f"Potential release of {package_name} {package_version}")  # noqa: T201
+        return True
+    else:
+        print(f"Skipped: {package_name} {package_version} already exists on PyPI")  # noqa: T201
+        return False
+
+
+def _check_no_dev_version(package_name, package_version):
+    if "dev" not in package_version:
+        return True
+    else:
+        print(f"Skipped: {package_name} {package_version} is still under development")  # noqa: T201
+        return False
+
+
+if __name__ == "__main__":
+    """Check if a package needs to be released"""
+    new_release = False
+    double_release = False
+    package_name = sys.argv[1]
+    package_version = sys.argv[2]
+
```

Comment:
if we do a check here similar check like:
```
    if len(sys.argv) != ARG_NUM:
        raise Exception("Usage: python extract_release_notes.py <filename> <heading>")
```

would the message be printed out if someone used it run when edit CICD process? if so, would it be helpful to have exception with usage instruction here as well